### ISSUE TITLE
Having issue with multiple database configs

### ISSUE THREAD

**Issue by clarkyx** [Appity/configature#1]
Configure seem to have trouble overriding alternative database configs within single config file

**DB config**
```
defaults: &defaults
  adapter: mysql2
  host: localhost
  encoding: utf8
  port: 3306
  username: root
  password:

development:
  <<: *defaults
  database: postageapp_dev

development_messages:
  <<: *defaults
  database: postageapp_messages_dev

test:
  <<: *defaults
  database: postageapp_test

test_messages:
  <<: *defaults
  database: postageapp_messages_test
```

**Configuare**
```
  RAILS_ENV = ENV['RAILS_ENV'] || ENV['APP_ENV'] || 'development'

  namespace :database do |db|
    db.env :RAILS_ENV, default: RAILS_ENV

    db.adapter default: 'mysql2'
    db.host default: 'localhost'
    db.port default: 3306, as: :integer
    db.username default: 'postage'
    db.password
    db.database default: 'postageapp_dev'
    db.max_connections default: 10, as: :integer
  end

  namespace :database_messages, env_suffix: '_messages', extends: :database do |db|
    db.database default: 'postageapp_messages_dev'
  end
```

When using `RAILS_ENV = 'test'`, will generates following result
```
#<Configature::Data adapter="mysql2", host="localhost", port=3306, username="root", password=nil, database="postageapp_test", max_connections=10>
#<Configature::Data adapter="mysql2", host="localhost", port=3306, username="postage", password=nil, database="postageapp_messages_dev", max_connections=10>
```

**Comment by tadman** [comment]
I've had a look at this and expanded the test coverage. It should work, but there are some tricks that need better documentation.

**Comment by tadman** [comment]
Fixed the source file problem where it was consistently looking in the wrong place.

Patched in 0.3.7.
